### PR TITLE
Deployment preparation: `AvalancheRootGaugeFactory` V2

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `AvalancheRootGaugeFactory` V2

Build-info: https://github.com/balancer/balancer-v2-monorepo/commit/227683919a7031615c0bc7f144666cdf3883d212 (master branch)

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [x] Task `README` has a description and links to the artifacts
- [x] `build-info` is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from `build-info`
- [x] New task has a working fork test 

## Deprecating old tasks checklist: 

- [x] Old task was moved to `deprecated/` directory
- [x] Deployments `README` is updated: old task is moved to the 'deprecated' section, and links are updated with the `deprecated/` prefix
- [x] Old task imports in `index.ts`, `input.ts` and fork tests are updated
- [x] Old task `README` is updated with a warning sign, a link to the replacement task and a short description 

## Code checklist:

- [x] The diff is legible and has no extraneous changes 
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -196,3 +197,4 @@ Go to each deprecated deployment's readme file to learn more about why it is dep
 | Linear Pools for Silo Tokens                        | [`20230315-silo-linear-pool`](./tasks/deprecated/20230315-silo-linear-pool)                             |
 | Composable Stable Pools V4                          | [`20230320-composable-stable-pool-v4`](./tasks/deprecated/20230320-composable-stable-pool-v4)           |
 | L2 Gauge Checkpointer                               | [`20230527-l2-gauge-checkpointer`](./tasks/deprecated/20230527-l2-gauge-checkpointer)                   |
+| Avalanche Root Gauge, for veBAL votin               | [`20230529-avalanche-root-gauge-factory`](./tasks/deprecated/20230529-avalanche-root-gauge-factory)     |
```
Comment (EndymionJkb) [inline README.md:200]:
```suggestion
| Avalanche Root Gauge, for veBAL voting              | [`20230529-avalanche-root-gauge-factory`](./tasks/deprecated/20230529-avalanche-root-gauge-factory)     |
```

**Review by EndymionJkb — APPROVED** [review]
Verified the build-info matches, and the fork test is equivalent to the old one (with more tests for the expanded functions). Just have a few suggestions

**Comment by jubeira** [comment]
Thanks for the review! Comments addressed in https://github.com/balancer/balancer-deployments/pull/76/commits/70969af7b9c0a58b32e3c6f01030eef982270cff

File: src/helpers/contracts/ILayerZeroBALProxy.sol [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+/**
+ * @dev Partial interface for LayerZero BAL proxy.
+ */
+interface ILayerZeroBALProxy {
+    /**
+     * @dev Emitted when `_amount` tokens are moved from the sender `_from` to (`_dstChainId`, `_toAddress`)
+     */
```
Comment (EndymionJkb) [inline src/helpers/contracts/ILayerZeroBALProxy.sol:24]:
```suggestion
    /// @dev Emitted when `_amount` tokens are moved from the sender `_from` to (`_dstChainId`, `_toAddress`)
```
One-liner :) Igor has convinced me this is good for those :)

**Review by EndymionJkb — APPROVED** [review]

**Review by EndymionJkb — APPROVED** [review]

**Pull request merged**